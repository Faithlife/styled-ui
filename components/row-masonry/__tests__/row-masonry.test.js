import { AutoSizedRowMasonry, MasonryCell, MasonryRow, getRowLayout } from '../';

describe('Row Masonry', () => {
	it('should be truthy', () => {
		expect.hasAssertions();

		expect(AutoSizedRowMasonry).toBeTruthy();
		expect(MasonryRow).toBeTruthy();
		expect(MasonryCell).toBeTruthy();
		expect(getRowLayout).toBeTruthy();
	});

	it('should return consistent row layouts', () => {
		const items = [
			{ aspectRatio: 2.21955943409412 },
			{ aspectRatio: 0.9859003797276708 },
			{ aspectRatio: 2.2994515698612124 },
			{ aspectRatio: 1.3756087709650013 },
			{ aspectRatio: 1.7468094204054432 },
			{ aspectRatio: 1.287234360985344 },
			{ aspectRatio: 0.7694801207159913 },
			{ aspectRatio: 0.6343636184394491 },
			{ aspectRatio: 0.9485378396801197 },
			{ aspectRatio: 2.082402503272989 },
			{ aspectRatio: 0.6632252433090523 },
			{ aspectRatio: 1.2269313864337024 },
			{ aspectRatio: 1.3638050079390163 },
			{ aspectRatio: 2.2283917834546902 },
			{ aspectRatio: 2.047257867779835 },
			{ aspectRatio: 1.0734050352965245 },
			{ aspectRatio: 2.074352961337868 },
			{ aspectRatio: 1.7873129356076118 },
			{ aspectRatio: 0.9430428677741222 },
			{ aspectRatio: 1.853901351550863 },
		];
		const getItemAspectRatio = jest.fn(x => x.aspectRatio);
		const layout = getRowLayout(items, {
			getItemAspectRatio,
			width: 300,
			gapWidth: 12,
			targetHeight: 100,
			minRowItems: 1,
			maxRowItems: 4,
		});

		expect(getItemAspectRatio).toHaveBeenCalled();
		expect(layout).toMatchSnapshot();
	});
});
