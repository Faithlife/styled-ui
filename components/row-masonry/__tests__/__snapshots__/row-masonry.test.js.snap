// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Masonry should return consistent row layouts 1`] = `
Object {
  "averageItemsPerRow": 2,
  "averageRowHeight": 96.17699655587734,
  "rows": Array [
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 2.21955943409412,
          "item": Object {
            "aspectRatio": 2.21955943409412,
          },
        },
        Object {
          "aspectRatio": 0.9859003797276708,
          "item": Object {
            "aspectRatio": 0.9859003797276708,
          },
        },
      ],
      "rowHeight": 89.84670428815163,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 2.2994515698612124,
          "item": Object {
            "aspectRatio": 2.2994515698612124,
          },
        },
        Object {
          "aspectRatio": 1.3756087709650013,
          "item": Object {
            "aspectRatio": 1.3756087709650013,
          },
        },
      ],
      "rowHeight": 78.36606022508269,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 1.7468094204054432,
          "item": Object {
            "aspectRatio": 1.7468094204054432,
          },
        },
        Object {
          "aspectRatio": 1.287234360985344,
          "item": Object {
            "aspectRatio": 1.287234360985344,
          },
        },
      ],
      "rowHeight": 94.92282272472104,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 0.7694801207159913,
          "item": Object {
            "aspectRatio": 0.7694801207159913,
          },
        },
        Object {
          "aspectRatio": 0.6343636184394491,
          "item": Object {
            "aspectRatio": 0.6343636184394491,
          },
        },
        Object {
          "aspectRatio": 0.9485378396801197,
          "item": Object {
            "aspectRatio": 0.9485378396801197,
          },
        },
      ],
      "rowHeight": 117.32790397747515,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 2.082402503272989,
          "item": Object {
            "aspectRatio": 2.082402503272989,
          },
        },
        Object {
          "aspectRatio": 0.6632252433090523,
          "item": Object {
            "aspectRatio": 0.6632252433090523,
          },
        },
      ],
      "rowHeight": 104.89404485313915,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 1.2269313864337024,
          "item": Object {
            "aspectRatio": 1.2269313864337024,
          },
        },
        Object {
          "aspectRatio": 1.3638050079390163,
          "item": Object {
            "aspectRatio": 1.3638050079390163,
          },
        },
      ],
      "rowHeight": 111.1653044383668,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 2.2283917834546902,
          "item": Object {
            "aspectRatio": 2.2283917834546902,
          },
        },
        Object {
          "aspectRatio": 2.047257867779835,
          "item": Object {
            "aspectRatio": 2.047257867779835,
          },
        },
      ],
      "rowHeight": 67.35818495251233,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 1.0734050352965245,
          "item": Object {
            "aspectRatio": 1.0734050352965245,
          },
        },
        Object {
          "aspectRatio": 2.074352961337868,
          "item": Object {
            "aspectRatio": 2.074352961337868,
          },
        },
      ],
      "rowHeight": 91.49369179839486,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 1.7873129356076118,
          "item": Object {
            "aspectRatio": 1.7873129356076118,
          },
        },
        Object {
          "aspectRatio": 0.9430428677741222,
          "item": Object {
            "aspectRatio": 0.9430428677741222,
          },
        },
      ],
      "rowHeight": 105.4807580914151,
    },
    Object {
      "gapWidth": 12,
      "items": Array [
        Object {
          "aspectRatio": 1.853901351550863,
          "item": Object {
            "aspectRatio": 1.853901351550863,
          },
        },
      ],
      "rowHeight": 100.91449020951458,
    },
  ],
}
`;
